tests/30rooms/11leaving.pl

```diff
@@ -0,0 +1,272 @@
+            )->on_done(sub { pass "User B set the room name the first time" })
+        })->then( sub {
+            $do_request_json_for->( $user_b,
+                method => "PUT",
+                uri => "/rooms/$room_id/state/madeup.test.state",
+                content => { "body" => "S1. B's state before A left", },
+            )->on_done(sub { pass "User B set the state the first time" })
+        })->then( sub {
+            $do_request_json_for->( $user_b,
+                method => "POST",
+                uri => "/rooms/$room_id/send/m.room.message",
+                content => {
+                    "body" => "M1. B's message before A left",
+                    "msgtype" => "m.room.text",
+                },
+            )->on_done(sub { pass "User B sent their first message" })
+        })->then( sub {
+            $do_request_json_for->( $user_b,
+                method => "POST",
+                uri => "/rooms/$room_id/send/m.room.message",
+                content => {
+                    "body" => "M2. B's message before A left",
+                    "msgtype" => "m.room.text",
+                },
+            )->on_done(sub { pass "User B sent their second message" })
+        })->then( sub {
+            $do_request_json_for->( $user_a,
+                method => "POST",
+                uri => "/rooms/$room_id/leave",
+                content => {},
+            )->on_done(sub { pass "User A left the room" })
+        })->then( sub {
+            $do_request_json_for->( $user_b,
+                method => "POST",
+                uri => "/rooms/$room_id/send/m.room.message",
+                content => {
+                    "body" => "M3. B's message after A left",
+                    "msgtype" => "m.room.text",
+                },
+            )->on_done(sub { pass "User B sent their third message" })
+        })->then( sub {
+            $do_request_json_for->( $user_b,
+                method => "PUT",
+                uri => "/rooms/$room_id/state/m.room.name",
+                content => { "name" => "N2. B's room name after A left", },
+            )->on_done(sub { pass "User B set the room name the second time" })
+        })->then( sub {
+            $do_request_json_for->( $user_b,
+                method => "PUT",
+                uri => "/rooms/$room_id/state/madeup.test.state",
+                content => { "body" => "S2. B's state after A left", },
+            )->on_done(sub { pass "User B set the state the second time" })
+        })
+    };
+
+
+test "A departed room is still included in /initialSync (SPEC-216)",
+    requires => [qw(do_request_json departed_room_id)],
+    check => sub {
+        my ($do_request_json, $departed_room_id) = @_;
+
+        $do_request_json->(
+            method => "GET",
+            uri => "/initialSync",
+            params => { limit => 2 },
+        )->then( sub {
+            my ( $body ) = @_;
```

Comment:
As there's a lot of `die` calls later on, it might be nice to future developers to put a call to
    log_if_fail "initialSync data", $body;

here so that if it dies, they can see what the `$body` data was.